packages/backend/src/storage/storage.ts

```diff
@@ -294,17 +296,29 @@ export class Storage {
     if (repo && !repo.eventsAttached) {
       log('Subscribing to channel ', channelData.address)
 
-      db.events.on('write', (_address, entry) => {
+      db.events.on('write', async (_address, entry) => {
+        const message = entry.payload.value
+        const crypto = getCrypto()
+        const signature = stringToArrayBuffer(message.signature)
+        const cryptoKey = await keyObjectFromString(message.pubKey, crypto)
+        const verify = await verifySignature(signature, message.message, cryptoKey)
         log(`Writing to public channel db ${channelData.address}`)
         this.io.loadMessages({
-          messages: [entry.payload.value]
+          messages: [entry.payload.value],
+          verifyStatus: verify
         })
       })
 
-      db.events.on('replicate.progress', (address, _hash, entry, progress, total) => {
+      db.events.on('replicate.progress', async (address, _hash, entry, progress, total) => {
```

Comment:
I suggest adding tests for those 2 events - verified and unverified message. We already mock some events in test, you can look at "Certificates and peers list are updated on replicated event" unit test.